### PR TITLE
Change execution types in sdk and deployedWallet

### PR DESCRIPTION
**Pull request by cehali**
# Summary
Change execution types in sdk and deployedWallet

### CODE REVIEW

File: universal-login-commons/lib/core/models/message.ts [hunk 1]
```diff
@@ -19,6 +19,15 @@ export type PaymentOptions = {
...
 };
 
+export type ExecutionOptions = {
+  gasPrice: utils.BigNumberish;
+  gasLimit?: utils.BigNumberish;
+  gasToken?: string;
+  nonce?: string | number;
+};
+
+export type SdkExecutionOptions = ExecutionOptions & Required<Pick<ExecutionOptions, 'gasLimit'>>;
```
Comment (xitronix) [inline universal-login-commons/lib/core/models/message.ts]: Can be PartialRequired<ExecutionOptions. 'gasLimit' | 'gasPrice'> (I checked but up to you what is more readable)

File: universal-login-react/src/ui/BackupCodes/BackupCodes.tsx [hunk 2]
```diff
@@ -28,7 +27,7 @@ export const BackupCodes = ({deployedWallet, basePath = '', className}: BackupPr
   const generateBackupCodes = async () => {
     try {
       history.replace(join(basePath, 'waitingForBackupCodes'));
-      const {waitToBeSuccess, waitForTransactionHash} = await deployedWallet.generateBackupCodes({...transactionDetails, ...gasParameters});
+      const {waitToBeSuccess, waitForTransactionHash} = await deployedWallet.generateBackupCodes(gasParameters!);
```
Comment (xitronix) [inline universal-login-react/src/ui/BackupCodes/BackupCodes.tsx]: I think we should use ensureNotNull before using "!"

File: universal-login-sdk/lib/api/DeployedWallet.ts [hunk 3]
```diff
@@ -39,34 +41,35 @@ export class DeployedWallet implements ApplicationWallet {
...
     return this.sdk.denyRequests(this.contractAddress, this.privateKey);
   }
 
-  async setRequiredSignatures(requiredSignatures: number, transactionDetails: Partial<Message>): Promise<Execution> {
-    return this.sdk.setRequiredSignatures(this.contractAddress, requiredSignatures, this.privateKey, transactionDetails);
+  async setRequiredSignatures(requiredSignatures: number, executionOptions: ExecutionOptions): Promise<Execution> {
+    return this.sdk.setRequiredSignatures(this.contractAddress, requiredSignatures, this.privateKey, {gasLimit: DEFAULT_GAS_LIMIT, ...executionOptions});
   }
 
   async execute(message: Partial<Message>): Promise<Execution> {
```
Comment (xitronix) [inline universal-login-sdk/lib/api/DeployedWallet.ts]: Shouldn't be here ExecutionOptions?

File: universal-login-commons/lib/core/models/message.ts (hunk 1 above)
Comment (cehali) [reply universal-login-commons/lib/core/models/message.ts]: PartialRequired<ExecutionOptions. 'gasLimit' | 'gasPrice'> doesn't work as expected

File: universal-login-sdk/lib/api/DeployedWallet.ts (hunk 3 above)
Comment (cehali) [reply universal-login-sdk/lib/api/DeployedWallet.ts]: Execute calls sdk.execute function which is more complex than executionOptions e.g. it should get parameter `to`.

Furthermore, execute is not called by any other function from deployedWallet.

File: universal-login-react/src/ui/BackupCodes/BackupCodes.tsx (hunk 2 above)
Comment (cehali) [reply universal-login-react/src/ui/BackupCodes/BackupCodes.tsx]: 👍

**Comment by cehali** [comment]
@xitronix @Marik-D please review once again after resolving conflicts

**Pull request merged**